src/main/java/teammates/ui/automated/PutCommentDocumentWorkerAction.java

```diff
@@ -0,0 +1,30 @@
+package teammates.ui.automated;
+
+import teammates.common.datatransfer.attributes.CommentAttributes;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+
+/**
+ * Task queue worker action: Puts searchable document for comment.
+ */
+public class PutCommentDocumentWorkerAction extends AutomatedAction {
+
+    @Override
+    protected String getActionDescription() {
+        return null;
+    }
+
+    @Override
+    protected String getActionMessage() {
+        return null;
+    }
+
+    @Override
+    public void execute() {
+        String commentId = getRequestParamValue(Const.ParamsNames.COMMENT_ID);
+        Assumption.assertNotNull(commentId);
+
+        CommentAttributes comment = logic.getComment(Long.valueOf(commentId));
```

Comment:
I do not understand the logic here:
What you did was to use the commentID to retrieve the comment from DB and put it back to DB, which I believe does no work. In InstructorStudentCommentAddAction.java, the **createdComment** is the thing that you are supposed to put into the DB.  However, u deleted the part that put createdComment into the DB and scheduled it into the task queue with CommentId. 
If ```logic.getComment(Long.valueOf(commentId))``` does not return null, you are just querying the object and put it back to DB without any updates. If it returns null, the comment will not be put into the DB.

I think what you should do is  to pass the ```CommentAttributes comment = extractCommentData();``` into your **scheduleSearchableDocumentsProductionForComments** and build the comment in PutCommentDocumentWorkerAction, and then from here u can just do a **logic.putDocument**